### PR TITLE
Connect plugins

### PR DESCRIPTION
**Pull request by Pierregvx**
- Connect delete feature to plugin card 
- Connect Redirect feature to plugin-card
- Connect update feature to plugin card 

- create mirage.js methods to mimic BE (delete, update)
- Update type on usePost.ts

### CODE REVIEW

File: web/massastation/src/pages/store/MyStation.tsx [hunk 1]
```diff
@@ -8,44 +8,58 @@ import { MyPlugin } from './MyPlugin';
...
 export function MyStation() {
   const navigate = useNavigate();
+
   const {
     error,
-    data: plugins,
+    data: myPlugins,
     isLoading,
+    // We fetch the plugin list everytime delete
+    refetch: refetchPlugins,
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply web/massastation/src/pages/store/MyStation.tsx:29]: what do you suggest, @Pierregvx  and I are not sure about our solution

File: web/massastation/src/mirage/server.js [hunk 2]
```diff
@@ -49,32 +44,37 @@ function mockServer(environment = ENV.DEV) {
...
+        const { id } = request.params;
+        let plugin = schema.plugins.find(id);
+        if (!plugin)
+          return new Response(404, {}, { code: '404', error: 'Not Found' });
+        return plugin.attrs;
+      });
 
       this.post('/plugin-manager/:id/execute', (schema, request) => {
-        const { method } = JSON.parse(request.requestBody);
+        const { command } = JSON.parse(request.requestBody);
```
[the remark this answers is not in the document]
Comment (Pierregvx) [reply web/massastation/src/mirage/server.js:63]: command is the name used in the back
https://github.com/massalabs/station/blob/37fa362b1bf83960521da917c9d749a89e85f3fb/api/swagger/server/restapi/resource/swagger.yml#L619
I did a mistake naming it method in a previous pr

File: web/massastation/src/pages/store/MyStation.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pierregvx) [reply web/massastation/src/pages/store/MyStation.tsx:28]: To clarify, It is called after a delete, we only refresh plugin-manager/id after an update
we can talk about it maybe

[the remark this answers is not in the document]
Comment (mazmassa) [reply web/massastation/src/pages/store/MyStation.tsx:28]: I guess the query already manages the refetch alone with no other action. But I need more context on that. An even better solution would be the BE send an event for us. But this is not possible now. 

So, keep this refetch after the delete and we check that together after!

File: web/massastation/src/custom/api/usePost.ts [hunk 3]
```diff
@@ -6,15 +6,15 @@ import { UseMutationResult, useMutation } from '@tanstack/react-query';
 
 // LOCALS
 
-export function usePost<T, P>(
+export function usePost<T>(
```
Comment (mazmassa) [inline web/massastation/src/custom/api/usePost.ts:9]: Please, open a new MR for the rest of them. useDelete, usePut, etc... All should follow the same. No need in this MR but we need update the others too.

BTW, in the wallet repo too!

File: web/massastation/src/mirage/server.js [hunk 4]
```diff
@@ -49,32 +44,39 @@ function mockServer(environment = ENV.DEV) {
...
-
         const plugin = schema.plugins.find(id);
 
         if (!plugin)
           return new Response(404, {}, { code: '404', error: 'Not Found' });
 
-        const status = ['update', 'start'].includes(method) ? 'Up' : 'Down';
-        const updatable = method === 'update' ? false : plugin.updatable;
+        const status = ['update', 'start'].includes(command) ? 'Up' : 'Down';
+        const updatable = command === 'update' ? false : plugin.updatable;
```
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:73]:
```suggestion
        const status = ['update', 'start'].includes(action) ? 'Up' : 'Down';
        const updatable = action === 'update' ? false : plugin.updatable;
```

File: web/massastation/src/mirage/server.js [hunk 5]
```diff
@@ -49,32 +44,39 @@ function mockServer(environment = ENV.DEV) {
...
+
+        if (!plugin)
+          return new Response(404, {}, { code: '404', error: 'Not Found' });
+
+        return plugin.attrs;
+      });
 
       this.post('/plugin-manager/:id/execute', (schema, request) => {
-        const { method } = JSON.parse(request.requestBody);
+        const { command } = JSON.parse(request.requestBody);
```
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:65]:
```suggestion
        const { action } = JSON.parse(request.requestBody);
```

Comment (pivilartisant) [reply web/massastation/src/mirage/server.js:65]: https://github.com/massalabs/station/pull/774#discussion_r1233764189

File: web/massastation/src/main.tsx [hunk 6]
```diff
@@ -18,8 +18,9 @@ import mockServer from './mirage/server.js';
 
 const baseURL = import.meta.env.VITE_BASE_APP;
 
-if ([ENV.DEV, ENV.TEST].includes(baseURL)) {
-  mockServer(baseURL);
+// Add ENV.STANDALONE to the array to enable MirageJS
+if ([ENV.DEV, ENV.TEST].includes(import.meta.env.VITE_ENV)) {
+  mockServer(import.meta.env.VITE_ENV);
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply web/massastation/src/main.tsx]: done

File: web/massastation/src/custom/api/usePost.ts (hunk 3 above)
Comment (pivilartisant) [reply web/massastation/src/custom/api/usePost.ts]: done

**Comment by pivilartisant** [comment]
We move the getRessources to the top level store.tsx to be passed as args to MyStation Section and Store section

File: web/massastation/src/mirage/server.js (hunk 4 above)
Comment (mazmassa) [reply web/massastation/src/mirage/server.js:73]: Ohh my bad! Got it! Well done! Thanks!

**Review by mazmassa — APPROVED** [review]

**Pull request merged**